addons/web/models/models.py

```diff
@@ -302,67 +468,37 @@ def group_id_name(value):
 
-        # get filter_values
-        filter_values = []
-
-        if field.type == 'many2one':
-            counters = {}
-            if not disable_counters:
-                groups = self.read_group(model_domain, [field_name], [field_name])
-                counters = {
-                    group[field_name][0]: group[field_name + '_count']
-                    for group in groups
-                }
-            # retrieve all possible values, and return them with their label and counter
-            field_names = ['display_name', group_by] if group_by else ['display_name']
-            records = Comodel.search_read(comodel_domain, field_names)
-            for record in records:
-                record_id = record['id']
-                values = {
-                    'id': record_id,
-                    'name': record['display_name'],
-                    'count': counters.get(record_id, 0),
-                }
-                if group_by:
-                    values['group_id'], values['group_name'] = group_id_name(record[group_by])
-                filter_values.append(values)
-
-        elif field.type == 'many2many':
-            # retrieve all possible values, and return them with their label and counter
-            field_names = ['display_name', group_by] if group_by else ['display_name']
-            records = Comodel.search_read(comodel_domain, field_names)
-            for record in records:
-                record_id = record['id']
-                values = {
-                    'id': record_id,
-                    'name': record['display_name'],
-                    'count': 0,
-                }
-                if not disable_counters:
-                    count_domain = AND([model_domain, [(field_name, 'in', record_id)]])
-                    values['count'] = self.search_count(count_domain)
-                if group_by:
-                    values['group_id'], values['group_name'] = group_id_name(record[group_by])
-                filter_values.append(values)
-
-        elif field.type == 'selection':
-            counters = {}
-            if not disable_counters:
-                groups = self.read_group(model_domain, [field_name], [field_name])
-                counters = {
-                    group[field_name]: group[field_name + '_count']
-                    for group in groups
-                }
-            # retrieve all possible values, and return them with their label and counter
-            selection = self.fields_get([field_name])[field_name]
-            for value, label in selection:
-                filter_values.append({
-                    'id': value,
-                    'name': label,
-                    'count': counters.get(value, 0),
-                })
-
-        return filter_values
+            def process_groupby_field(record):
+                value = record[group_by]
+                record['group_id'], record['group_name'] = group_id_name(value)
+                del record[group_by]
```

Comment:
You can use `pop` to remove and return the element:
```suggestion
                value = record.pop(group_by)
                record['group_id'], record['group_name'] = group_id_name(value)
```

Reply:
ok